packages/jsii-rosetta/bin/jsii-rosetta.ts

```diff
@@ -144,7 +160,11 @@ function main() {
           );
         }
 
-        if (result.diagnostics.some(isErrorDiagnostic) && args.fail) {
+        if (
+          result.diagnostics.some((diag) =>
+            isErrorDiagnostic(diag, { onlyStrict: !args.fail }),
+          )
+        ) {
           process.exit(1);
```

Comment:
Not your change, but since you need to do a pass anyway:

This needs to be:

```suggestion
          process.exitCode = 1;
```